tests/mock_crust/client_restrictions.rs

```diff
@@ -118,3 +120,49 @@ fn reconnect_disconnected_client() {
     let _ = poll_all(&mut nodes, &mut clients);
     expect_next_event!(unwrap!(clients.last_mut()), Event::Connected);
 }
+
+
+/// Confirming the number of user message parts being sent in case of exceeding limit.
+#[test]
+fn resending_parts_on_exceeding_limit() {
+    let network = Network::new(MIN_SECTION_SIZE, None);
+    let mut rng = network.new_rng();
+    let mut nodes = create_connected_nodes(&network, MIN_SECTION_SIZE);
+    let mut clients = create_connected_clients(&network, &mut nodes, 1);
+
+    let mut datas = vec![
```

Comment:
the plural for data is data itself - singular would be datum but rarely used. Maybe data_coll for collection then `for data in data_coll {` below in the for loop ?